### ISSUE TITLE
cannot run keras test autoencoders code:  Error when checking model target

### ISSUE THREAD

**Issue by drmeerkat** [fchollet/keras#5324]
I have update my keras, tf to the newest version and I'm using win10+vs2015
I tried to run the code in [https://blog.keras.io/building-autoencoders-in-keras.html](url)
I have tested all the other encoders like the 'Deep autoencoder'
But when it comes to modules including Convolution2D layers, the test code will throw the error:
ValueError 

> Error when checking model target: expected convolution2d_5 to have shape (None, 4, 28, 1) but got array with shape (60000, 1, 28, 28)

And here is the code (i just copy them from the keras blog)(this is the 'Application to image denoising')
`from keras.datasets import mnist`
`import numpy as np`
`(x_train, _), (x_test, _) = mnist.load_data('D:\mnist.pkl.gz')`

`x_train = x_train.astype('float32') / 255.`
`x_test = x_test.astype('float32') / 255.`
`x_train = np.reshape(x_train, (len(x_train), 1, 28, 28))`
`x_test = np.reshape(x_test, (len(x_test), 1, 28, 28))`
`print(x_train.shape)`
`print(x_test.shape)`

`noise_factor = 0.5`
`x_train_noisy = x_train + noise_factor * np.random.normal(loc=0.0, scale=1.0, size=x_train.shape) `
`x_test_noisy = x_test + noise_factor * np.random.normal(loc=0.0, scale=1.0, size=x_test.shape) `

`x_train_noisy = np.clip(x_train_noisy, 0., 1.)`
`x_test_noisy = np.clip(x_test_noisy, 0., 1.)`

`from keras.layers import Input, Dense, Convolution2D, MaxPooling2D, UpSampling2D, Reshape`
`from keras.models import Model`

`input_img = Input(shape=(1, 28, 28))`

`x = Convolution2D(32, 3, 3, activation='relu', border_mode='same')(input_img)`
`x = MaxPooling2D((2, 2), border_mode='same')(x)`
`x = Convolution2D(32, 3, 3, activation='relu', border_mode='same')(x)`
`encoded = MaxPooling2D((2, 2), border_mode='same')(x)`

`# at this point the representation is (32, 7, 7)`

`x = Convolution2D(32, 3, 3, activation='relu', border_mode='same')(encoded)`
`x = UpSampling2D((2, 2))(x)`
`x = Convolution2D(32, 3, 3, activation='relu', border_mode='same')(x)`
`x = UpSampling2D((2, 2))(x)`
`decoded = Convolution2D(1, 3, 3, activation='sigmoid', border_mode='same')(x)`

`autoencoder = Model(input_img, decoded)`
`autoencoder.compile(optimizer='adadelta', loss='binary_crossentropy')`
`
print('ok')
from keras.callbacks import TensorBoard`

`autoencoder.fit(x_train_noisy, x_train,`
              `nb_epoch=100,`
              `batch_size=128,`
              `shuffle=True,`
              `validation_data=(x_test_noisy, x_test),`
              `callbacks=[TensorBoard(log_dir='/tmp/tb', histogram_freq=0, write_graph=False)])`

**Comment by drmeerkat** [comment]
I found that the code given in the blog is using th as backend.
So just change 
`
x_train = np.reshape(x_train, (len(x_train), 1, 28, 28))`
`x_test = np.reshape(x_test, (len(x_test), 1, 28, 28))
`
to 
`
x_train = np.reshape(x_train, (len(x_train), 28, 28,1))`
`x_test = np.reshape(x_test, (len(x_test), 28, 28, 1))
`

can fix this perfectly
you can also check [here](https://github.com/fchollet/keras/issues/4036)
